reader_concurrency_semaphore.cc

```diff
@@ -864,35 +864,58 @@ future<> reader_concurrency_semaphore::enqueue_waiter(reader_permit permit, read
 }
 
 void reader_concurrency_semaphore::evict_readers_in_background() {
+    if (_evicting) {
+        return;
+    }
+    _evicting = true;
     // Evict inactive readers in the background while wait list isn't empty
     // This is safe since stop() closes _gate;
```

Comment:
The condition below in the eviction loop is `do_until([this] { return _wait_list.empty() || _inactive_reads.empty(); },`
We'll stop as soon as the wait list is empty. Note that this might still be too aggressive; wince the new admission algorithm, we  might not admit new reads even when memory is available, because all not current reads are blocked (some need more CPU). We could also add this to the condition below: evict only if there is a waiter blocked on resources.

Reply:
Opened https://github.com/scylladb/scylladb/issues/11803 for the too aggressive eviction.